aea/protocols/generator.py

```diff
@@ -278,52 +295,60 @@ def _message_class_str(self) -> str:
-        )
-        cls_str += "            assert isinstance(\n"
-        cls_str += "                self.dialogue_reference[0], str\n"
-        cls_str += "            ), \"The first element of dialogue_reference must be 'str' but it is not.\"\n"
-        cls_str += "            assert isinstance(\n"
-        cls_str += "                self.dialogue_reference[1], str\n"
-        cls_str += "            ), \"The second element of dialogue_reference must be 'str' but it is not.\"\n"
+        cls_str += '            assert type(self.dialogue_reference) == tuple, "dialogue_reference must be \'tuple\' but it is not."\n'
+        cls_str += '            assert type(self.dialogue_reference[0]) == str, "The first element of dialogue_reference must be \'str\' but it is not."\n'
+        cls_str += '            assert type(self.dialogue_reference[1]) == str, "The second element of dialogue_reference must be \'str\' but it is not."\n'
         cls_str += (
             '            assert type(self.message_id) == int, "message_id is not int"\n'
         )
         cls_str += '            assert type(self.target) == int, "target is not int"\n'
-        cls_str += '            assert type(self.performative) == str, "performative is not str"\n\n'
+        cls_str += '            assert type(self.performative_raw) == str, "performative is not str"\n\n'
 
         cls_str += "            # Light Protocol 2\n"
         cls_str += "            # Check correct performative\n"
         cls_str += "            assert (\n"
-        cls_str += "                self.performative in self.valid_performatives\n"
-        cls_str += "            ), \"'{}' is not in the list of valid performativs: {}\".format(\n"
-        cls_str += "                self.performative, self.valid_performatives\n"
+        cls_str += "                self.performative_raw in self.valid_performatives\n"
+        cls_str += "            ), \"'{}' is not in the list of valid performatives: {}\".format(\n"
+        cls_str += "                self.performative_raw, self.valid_performatives\n"
         cls_str += "            )\n\n"
         cls_str += "            # Check correct contents\n"
         cls_str += (
             "            actual_nb_of_contents = len(self.body) - DEFAULT_BODY_SIZE\n"
         )
+        counter = 1
         for performative, contents in all_contents.items():
-            cls_str += '            if self.performative == "{}":\n'.format(
-                performative
-            )
-            cls_str += "                expexted_nb_of_contents = {}\n".format(
-                len(contents)
-            )
-            if len(contents) == 0:
-                continue
-            for content_name, content_type in contents.items():
-                cls_str += '                assert type(self.{}) == {}, "{} is not {}"\n'.format(
-                    content_name, content_type, content_name, content_type
+            if counter == 1:
```

Comment:
you can rewrite the next ~20 lines because the if, else only applies to the first line `if, elif` part, all else is the same

Reply:
Done